src/Symfony/Component/Console/Application.php

```diff
@@ -88,8 +89,8 @@ public function __construct(string $name = 'UNKNOWN', string $version = 'UNKNOWN
         $this->version = $version;
         $this->terminal = new Terminal();
         $this->defaultCommand = 'list';
-        $this->signalRegistry = new SignalRegistry();
-        if (\defined('SIGINT')) {
+        if (\defined('SIGINT') && SignalRegistry::isSupported()) {
+            $this->signalRegistry = new SignalRegistry();
```

Comment:
Sorry for the late review, but the PR has changed before my review.
So here we go:

This change is wrong. See `getSignalRegistry()` method:
https://github.com/chalasr/symfony/blob/8fe876341e860ea3fa13bec3c2bdf77a66ead424/src/Symfony/Component/Console/Application.php#L111-L114

Then, a lot of fatal error could occurs since the registry might not be defined

Reply:
Oops, thank you; #38602 makes it nullable